scripts/reset-test-db.sh

```diff
@@ -28,34 +28,54 @@ DB_CACHE_DIR="common/test/db_cache"
 echo "CREATE DATABASE IF NOT EXISTS edxtest;" | mysql -u root
 
 # Clear out the test database
+#
+# We are using the django-extensions's reset_db command which uses "DROP DATABASE" and
+# "CREATE DATABASE" in case the tests are being run in an environment (e.g. devstack
+# or a jenkins worker environment) that already ran tests on another commit that had
+# different migrations that created, dropped, or altered tables.
+#
+# Note that as a side effect, this might render the --serversonly / --testsonly
+# paver options useless or broken.
```

Comment:
@jzoldak One possible solution I am thinking for this is:
1. When running with `--serversonly`, after this script and `BokChoyTestSuite.load_data()` have run, do 
```
# This dumps the initial state of the data into the json file.
./manage.py lms --settings bok_choy dumpdata > $SOME_TEMP_DB_CACHE_DIR/bok_choy_data.json
```

2. When running with `--testsonly`, do:
```
./manage.py lms --settings bok_choy flush --traceback --noinput  # This will empty all tables
./manage.py lms --settings bok_choy loaddata $SOM_TEMP_DB_CACHE_DIR/bok_choy_data.json # This will put back the initial data into the database
```